### ISSUE TITLE
Blazor server JSRuntime.Confirm() is causing page reload / connection-reset on waiting 10 minutes

### ISSUE THREAD

**Issue by jaishmathews** [dotnet/aspnetcore#42951]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I deployed a simple .NET Core 6 Blazor server app here https://epd-web.azurewebsites.net/ in Azure . A javascript confirm ( JSRuntime.Confirm("Please select")) will be prompted on clicking below button.

![image](https://user-images.githubusercontent.com/26008982/181346273-14c9e566-38ee-4221-bd78-5808030a5aaa.png)
If you wait for more than 10 minutes and then respond to "OK" button, "Attempting to Reconnect to the Server: 1 of 8" and then "Could not reconnect to the server. Reload the page to restore functionality." appears.

I already added below middleware settings.

builder.Services.AddServerSideBlazor(options =>
            {
                options.DetailedErrors = true;
                options.JSInteropDefaultCallTimeout = TimeSpan.FromHours(1);
            });

 app.UseEndpoints(endpoints =>
                // other settings go here
                endpoints.MapBlazorHub(options => options.WebSockets.CloseTimeout = new TimeSpan(1, 1, 1)));
[azure](https://stackoverflow.com/questions/tagged/azure)

I have a ticket here https://stackoverflow.com/questions/73123294/blazor-server-jsruntime-confirm-is-causing-page-reload-connection-reset-on-w as well for the same issue.

### Expected Behavior

On clicking "OK" button of the confirmation dialog, the execution flow should continues without any breaks.

### Steps To Reproduce

- Prompt a jsRuntime.InvokeAsync<bool>("confirm",CancellationToken.None, message);
- Wait for 12 minutes
- Then click OK button
- You see  "Attempting to Reconnect to the Server: 1 of 8" and then "Could not reconnect to the server. Reload the page to restore functionality." appears.

### .NET Version

6

**Comment by javiercn** [comment]
@jaishmathews thanks for contacting us.

Very likely that is blocking the UI, which prevents SignalR from receiving pings in a timely manner. If you want to display a dialog, I suggest you use the `<dialog>` HTML element instead.

**Comment by jaishmathews** [comment]
- I will configure  signalR client options and try
- Regarding <dialog>, it won't block the code execution flow as I understood. I need to block lines of code execution once dialog opened and based on "confirm" dialog response of "OK" / "Cancel", proceeding code to be executed. If I use <dialog>, dialog will open and proceeding lines of code also will be executed in parallel. .

**Comment by javiercn** [comment]
> * Regarding , it won't block the code execution flow as I understood. I need to block lines of code execution once dialog opened and based on "confirm" dialog response of "OK" / "Cancel", proceeding code to be executed. If I use , dialog will open and proceeding lines of code also will be executed in parallel. .

Using a blocking prompt like Confirm is something that won't work well with Blazor Server as you are preventing the heartbeat mechanism used to ensure the connection is alive from working correctly. It is also something that browsers discourage and in some cases block, so I don't think that it will lead to a good experience.

I am not 100% what you mean by the `dialog will open and proceeding lines of code also will be executed in parallel`. The dialog element allows you to trap the user focus and will prevent further interactions with the app outside of the dialog, other than anything that you have running in a timer or inside the server, both of which, you can control. That is just my recommendation.

**Comment by jaishmathews** [comment]
Ref. to below diagram - After red marked line executed for confirmation dialog, green marked lines and also this method calling side also sohuld wait till confirmation dialog get a response from use. Using a custom dialog won't do such waiting. That is whay I am telling "custom dialog will open and proceeding lines of code also will be executed in parallel". Otherwiese I need to move all such proceeding code directly under OK/Cancel button of the custom dialog. It's a huge effort at this moment.

![image](https://user-images.githubusercontent.com/26008982/181492448-59bde9e9-0a4c-4e59-b327-5f56c5302e82.png)


Meantime, I couldn't locate a timeout parameter to set using JS Blazor.start({. Any lead here?

**Comment by javiercn** [comment]
@jaishmathews you can do something like this:

```javascript
Blazor.start({
    configureSignalR: function (builder) {
        let c = builder.build();
        c.serverTimeoutInMilliseconds = 60000;
        c.keepAliveIntervalInMilliseconds = 30000;
        builder.build = () => {
            return c;
        };
    };
});
```

**Comment by jaishmathews** [comment]
Increased little more duration for timeout like below without any luck. Trying further options.

 <script src="_framework/blazor.server.js" autostart="false"></script>
    <script>
        Blazor.start({
            configureSignalR: function(builder) {
                let c = builder.build();
                c.serverTimeoutInMilliseconds = 3600000;
                c.keepAliveIntervalInMilliseconds = 30000;
                builder.build = () => {
                    return c;
                };
            }
        });
    </script>
